src/control/cmd/daos/flags.go

```diff
@@ -241,3 +241,34 @@ func (f *ContTypeFlag) UnmarshalFlag(fv string) error {
 	f.Set = true
 	return nil
 }
+
+type FsCheckFlag struct {
+	Set   bool
+	Flags C.uint64_t
+}
+
+func (f *FsCheckFlag) UnmarshalFlag(fv string) error {
+	if fv == "" {
+		return errors.New("empty filesystem check flags")
+	}
+
+	f.Flags = 0
+
+	cflags := strings.Split(fv, ",")
+	for _, cflag := range cflags {
+		if strings.EqualFold(cflag, "print") {
+			f.Flags |= C.DFS_CHECK_PRINT
+		} else if strings.EqualFold(cflag, "remove") {
+			f.Flags |= C.DFS_CHECK_REMOVE
+		} else if strings.EqualFold(cflag, "relink") {
+			f.Flags |= C.DFS_CHECK_RELINK
+		} else if strings.EqualFold(cflag, "verify") {
+			f.Flags |= C.DFS_CHECK_VERIFY
+		} else {
+			return errors.Errorf("unknown filesystem check flags: %q", fv)
+		}
+	}
```

Comment:
```suggestion
	for _, cflag := range strings.Split(fv, ",") {
		switch strings.TrimSpace(strings.ToLower(cflag)) {
		case "print":
			f.Flags |= C.DFS_CHECK_PRINT
		case "remove":
			f.Flags |= C.DFS_CHECK_REMOVE
		case "relink":
			f.Flags |= C.DFS_CHECK_RELINK
		case "verify":
			f.Flags |= C.DFS_CHECK_VERIFY
		default:
			return errors.Errorf("unknown filesystem check flags: %q", fv)
		}
	}
```
What you have isn't wrong, but it could be written more idiomatically. The only thing I'd strongly suggest fixing is that your approach doesn't handle untrimmed inputs.